gnovm/pkg/gnolang/values_string.go

```diff
@@ -164,8 +164,27 @@ func (v RefValue) String() string {
 func (tv *TypedValue) Sprint(m *Machine) string {
 	// if undefined, just "undefined".
 	if tv == nil || tv.T == nil {
-		return "undefined"
+		return undefinedStr
 	}
+
+	if tv.V == nil && (tv.T.Kind() == SliceKind || tv.T.Kind() == StringKind) {
+		return undefinedStr
+	}
+
+	if tv.T.Kind() == FuncKind {
```

Comment:
I don't think this is necessary.  The switch case below should handle this, right?